From bdf65393678544a857b588245b2d1f37f25f0acc Mon Sep 17 00:00:00 2001
From: ShevT <shev_t@mail.ru>
Date: Tue, 14 Mar 2017 11:41:38 +0300
Subject: [PATCH] fwb: Fix Sending non-protected broadcast PROXIMITY_CP and
 GRIPSENSOR_CP

This patch fixes errors:
E ActivityManager: Sending non-protected broadcast android.intent.action.PROXIMITY_CP from system uid 1001 pkg null
E ActivityManager: Sending non-protected broadcast android.intent.action.GRIPSENSOR_CP from system uid 1001 pkg null

project frameworks/base/
diff --git a/frameworks/base/core/java/android/content/Intent.java b/frameworks/base/core/java/android/content/Intent.java
index 1afae798bcf..e52e3bc6036 100644
--- a/frameworks/base/core/java/android/content/Intent.java
+++ b/frameworks/base/core/java/android/content/Intent.java
@@ -3324,6 +3324,28 @@ public class Intent implements Parcelable, Cloneable {
      */
     public static final int EXTRA_THERMAL_STATE_EXCEEDED = 2;
 
+    /**
+     * ACTIVITY INTENT ACTION PROXIMITY_CP, GRIPSENSOR_CP
+     */
+    /** {@hide} */
+    @SdkConstant(SdkConstantType.ACTIVITY_INTENT_ACTION)
+    public static final String PROXIMITY_CP = "android.intent.action.PROXIMITY_CP";
+
+    /** {@hide} */
+    @SdkConstant(SdkConstantType.ACTIVITY_INTENT_ACTION)
+    public static final String GRIPSENSOR_CP = "android.intent.action.GRIPSENSOR_CP";
+
+    /** {@hide} */
+    @SdkConstant(SdkConstantType.ACTIVITY_INTENT_ACTION)
+    public static final String PHONE_EXSTATE_CHANGED = "android.intent.action.PHONE_EXSTATE_CHANGED";
+
+    /** {@hide} */
+    @SdkConstant(SdkConstantType.ACTIVITY_INTENT_ACTION)
+    public static final String WB_AMR = "android.intent.action.WB_AMR";
+
+    /** {@hide} */
+    @SdkConstant(SdkConstantType.ACTIVITY_INTENT_ACTION)
+    public static final String pcmclkctrl = "android.intent.action.pcmclkctrl";
 
     // ---------------------------------------------------------------------
     // ---------------------------------------------------------------------
diff --git a/frameworks/base/core/res/AndroidManifest.xml b/frameworks/base/core/res/AndroidManifest.xml
index eddbc1d19a9..d371abad3a4 100644
--- a/frameworks/base/core/res/AndroidManifest.xml
+++ b/frameworks/base/core/res/AndroidManifest.xml
@@ -512,6 +512,13 @@
     <protected-broadcast android:name="android.intent.action.MANAGED_PROFILE_UNAVAILABLE" />
     <protected-broadcast android:name="com.android.server.pm.DISABLE_QUIET_MODE_AFTER_UNLOCK" />
 
+    <protected-broadcast android:name="android.intent.action.GRIPSENSOR_CP" />
+    <protected-broadcast android:name="android.intent.action.PROXIMITY_CP" />
+
+    <protected-broadcast android:name="android.intent.action.PHONE_EXSTATE_CHANGED" />
+    <protected-broadcast android:name="android.intent.action.WB_AMR" />
+    <protected-broadcast android:name="android.intent.action.pcmclkctrl" />
+
     <protected-broadcast android:name="android.telecom.action.TTY_PREFERRED_MODE_CHANGED" />
     <protected-broadcast android:name="android.telecom.action.CURRENT_TTY_MODE_CHANGED" />
     <protected-broadcast android:name="com.android.server.retaildemo.ACTION_RESET_DEMO" />
